File: java/leetcode_grind/Day602.java
package leetcode_grind;

import java.util.Stack;

public class Day602 {
    // https://leetcode.com/problems/crawler-log-folder/description/?envType=daily-question&envId=2024-07-10
    static class Solution1 {
        public int minOperations(String[] logs) {
            Stack<String> folderStack = new Stack<>();
            for (String currentOperation : logs) {
                if (currentOperation.equals("../")) {
                    if (!folderStack.isEmpty()) {
                        folderStack.pop();
                    }
                } else if (!currentOperation.equals("./")) {
                    folderStack.push(currentOperation);
                }
            }
            return folderStack.size();
        }
    }
}

File: src/leetcode_grind/day_602.rs
// https://leetcode.com/problems/crawler-log-folder/description/?envType=daily-question&envId=2024-07-10
pub fn min_operations_1(logs: Vec<String>) -> i32 {
    let mut folder_depth = 0;
    for current_operation in logs {
        if current_operation == "../" {
            folder_depth = 0.max(folder_depth - 1);
        } else if (current_operation != "./") {
            folder_depth += 1;
        }
    }
    folder_depth
}
